library/js/clickmap.js

```diff
@@ -1,140 +1,161 @@
-                                    container.append(marker);
-                                    if ( fn_isnumber(newcounter) ) { counter++; }
-                            }
-                            dialog.remove();
-                    };
-
-                    dialog.dialog({
-                            title: "Information",
-                            autoOpen: false, height: hasOptions? 345 : 300, width: 350, modal:true,
-                            open: function() { dialog.find(".detail").focus(); },
-                            buttons: {
-                                    "Save": function() { dialog.saved = true; $(this).dialog("close"); },
-                                    "Cancel": function() { $(this).dialog("close"); }
-                            },
-                            close: do_marker
-                    });
-                    dialog.dialog("open");
+                }
+            }
+
+            const do_marker = function() {
+                const newcounter = $('#counterInput').val();
+                const notes = encodeURIComponent($('#detailTextArea').val());
+                const selectedOption = encodeURIComponent($('#painScaleSelect').val());
+
+                let combinedNotes = "";
+                if (selectedOption) {
+                    combinedNotes = options[selectedOption];
+                }
+                if (selectedOption && notes) {
+                    combinedNotes += "%3A%20";
+                }
+                if (notes) {
+                    combinedNotes += notes;
+                }
+
+                const marker = fn_buildMarker(x, y, newcounter, combinedNotes);
+                container.append(marker);
+                if (fn_isnumber(newcounter)) {
+                    counter++;
+                }
+            };
+
+            dlgopen('', '', hasOptions? 345 : 300, 350, false, 'Information', {
+                buttons: [{
+                    text: 'Save',
+                    close: true,
+                    style: 'btn-sm btn-primary',
+                    click: do_marker,
+                }, {
+                    text: 'Cancel',
+                    close: true,
+                    style: 'btn-sm btn-secondary'}],
+                type: 'Alert',
+                html: `
+                    <div class="form-group">
+                        <label for="counterInput">Label</label>
+                        <input type="text" class="form-control" id="counterInput" value="${counter + 1}">
+                    </div>
+                    <div class="form-group">
+                        <label for="painScaleSelect">${optionsTitle}</label>
+                        <select class="form-control" id="painScaleSelect">
+                            ${optionsSelect}
+                        </select>
+                    </div>
+                    <div class="form-group">
+                        <label for="detailTextArea">Detail</label>
```

Comment:
translate. need to work in the xl() around `Detail` here (not sure how to do when in the ``` stuff)